discovery-provider/src/utils/prometheus_metric.py

```diff
@@ -1,26 +1,35 @@
 from time import time
-from typing import Dict
+from typing import Callable, Dict
 
-from prometheus_client import Histogram
+from prometheus_client import Gauge, Histogram
 
 
 class PrometheusMetric:
     histograms: Dict[str, Histogram] = {}
+    gauges: Dict[str, Gauge] = {}
+    registered_collectors: Dict[str, Callable] = {}
 
-    def __init__(self, name, description, labelnames=()):
+    def __init__(self, name, description, labelnames=(), gauge=False):
         self.reset_timer()
 
         # set metric prefix of audius_project_
         name = f"audius_dn_{name}"
 
         # CollectorRegistries must be uniquely named
-        if name not in PrometheusMetric.histograms:
-            # NOTE: we only set labelnames once.
-            # unsure if overloading is supported.
-            PrometheusMetric.histograms[name] = Histogram(
-                name, description, labelnames=labelnames
-            )
-        self.h = PrometheusMetric.histograms[name]
+        # NOTE: we only set labelnames once.
+        # unsure if overloading is supported.
+        if gauge:
+            if name not in PrometheusMetric.gauges:
+                PrometheusMetric.gauges[name] = Gauge(
```

Comment:
Phew! Thanks! I knew I didn't like the `gauge=false` bit since I would only extend to two types, but figured to punt that design choice until later.

I'll implement your suggestion, thanks!

Reply:
Heh, I misread your suggestion, but hopefully the latest change is what you had in mind?